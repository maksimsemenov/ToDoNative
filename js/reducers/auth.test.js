import authReducer, { logIn, logOut, getUser } from './auth'

describe('Auth reducer', () => {
  it('returns initial state', () => {
    expect(authReducer()).toEqual({})
  })
  it('returns state for action with unknown type', () => {
    expect(
      authReducer({ requesting: true }, { type: 'AUTH_UNKNOWN_tYPE' })
    ).toEqual({ requesting: true })
  })
  it('sets user for logIn action', () => {
    const state = { user: { displayName: 'Han Solo' } }
    expect(
      authReducer(state, logIn({ displayName: 'Luke Skywalker' }))
    ).toEqual({ user: { displayName: 'Luke Skywalker' } })
  })
  it('clears user for logOut action', () => {
    const state = { user: { displayName: 'Han Solo' }, requesting: true }
    expect(authReducer(state, logOut())).toEqual({ requesting: true })
  })
})

describe('getUser selector', () => {
  it('returns undefined for empty state', () => {
    expect(getUser({})).toBeUndefined()
  })
  it('returns undefined for empty auth state', () => {
    expect(getUser({ uiState: { loading: true } })).toBeUndefined()
  })
  it('returns correct user', () => {
    const state = {
      auth: {
        user: { displayName: 'Han Solo' }
      }
    }
    expect(getUser(state)).toMatchSnapshot()
  })
  it('returns the same user if unrelated part of state had changed', () => {
    const state = {
      uiState: { loading: true },
      auth: {
        user: { displayName: 'Han Solo' }
      }
    }
    const nextState = {
      uiState: { loading: false },
      auth: {
        ...state.auth,
        requesting: true
      }
    }
    expect(getUser(state)).toBe(getUser(nextState))
  })
  it('returns the same undefined user if unrelated part of state had changed', () => {
    const state = {
      uiState: { loading: true },
      auth: {}
    }
    const nextState = { ...state, uiState: { loading: false } }
    expect(getUser(state)).toBe(getUser(nextState))
  })
})
