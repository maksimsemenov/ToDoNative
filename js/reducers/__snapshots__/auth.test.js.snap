// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUser selector returns correct user 1`] = `
Object {
  "displayName": "Han Solo",
}
`;
